test/runtests.jl

```diff
@@ -63,8 +63,16 @@ eltypes = (Int64, Float64, BigFloat)
             end
             @test Matrix(add_qr(S, S)) ≈ Matrix(S) + Matrix(S)
             if (size(M, 1) >= size(M, 2))
-                @test choleskify_factor(S).R ≈ cholesky(Matrix(S)).U
                 @test Matrix(add_cholesky(S, S)) ≈ Matrix(S) + Matrix(S)
+
+                tri = triangularize_factor(S)
+                chol = cholesky(Matrix(S))
+                # Assert that `tri` is basically a Cholesky factorisation, 
+                # but without guaranteed positive diagonal elements
+                @test begin
+                    tri.R isa UpperTriangular
+                    tri.R' * tri.R ≈ chol.U' * chol.U
+                end
```

Comment:
This should actually suffice
```suggestion
                tri = triangularize_factor(S)
                @test tri.R isa UpperTriangular
```

Reply:
But this does not actually test if the matrix is still correct? E.g. a `triangularize_factor(S) = PSDMatrix(UpperTriangular(rand(2,2)))` would pass the test?